### DISCUSSION TITLE
How to change font size for all code in html output?

### DISCUSSION THREAD

**Discussion by royfrancis** [quarto-dev/quarto-cli#8005] [Q&A]
### Description

I want to change font size of all code in the html document. By all code elements, I mean unexecuted blocks, blocks with language (with highlighting), inline code, code output etc. All monospaced font I guess.

````
---
title: Code font
---

```
date()
```

```r
date()
```

```{r}
date()
```

This is the `date()`.

This is the `r date()`.

This is the `{r} date()`.
````

![Screenshot 2023-12-21 at 14 49 48](https://github.com/quarto-dev/quarto-cli/assets/5083500/9080fd4c-5c15-47d0-9368-a28fcb90411c)

I couldn't really find a variable for code font size [here](https://quarto.org/docs/output-formats/html-themes.html). Looking into the css, I see that there are many things to target: pre, code, sourceCode etc. Is there a good way to target the size of all code elements similar to `$font-family-monospace`?

`quarto 1.3.433`

**Comment by mcanouil** [comment]
monofont does not work?
It's from the link you shared.
![image](https://github.com/quarto-dev/quarto-cli/assets/8896044/ca2c9534-4da2-4852-9d16-0e0af55b0a4c)

Edit: I misread, font size is for all. You'll have to use your favourite browser inspect mode to target the proper elements and/or use the SCSS variables described in the link you shared to make a new rule.

**Comment by royfrancis** [comment]
I think something like this should target all of them.

```
pre, code, .sourceCode {
  font-size: 1.1rem;
}
```

But this won't look good if monospace/code is used in titles, callouts etc where font sizes are scaled differently. There is a discussion [here](https://stackoverflow.com/questions/75907436/quarto-revealjs-increase-relative-font-size-of-code-chunks).

Perhaps `em` units might be better?

```
pre, code, .sourceCode {
  font-size: 1.1em;
}
```

Update: This works for a my specific font and this is specifically for revealjs slides:

```
.reveal pre {
  font-size: 1.8rem !important;
}

.reveal code {
  font-size: 0.85em !important;
}
```

But these values will have to be adjusted for a new font.

**Comment by sorenwacker** [comment]
I would appreciate a more comprehensive answer. These code fragments without context are of little help.

**Reply by cderv** [reply]
The snippets above https://github.com/quarto-dev/quarto-cli/discussions/8005#discussioncomment-7921479 are some CSS rules to included with the document. 

They must be provided through CSS file (https://quarto.org/docs/output-formats/html-basics.html#css-styles) or using Quarto Theme feature. (https://quarto.org/docs/output-formats/html-themes.html#theme-options) 

Hope this helps understand the context for the answer above
